src/test/java/edu/kit/compiler/lexer/LexerTest.java

```diff
@@ -166,9 +171,10 @@ public void testZeroLiteral() throws LexException {
     }
 
     @Test
-    public void testIllegalLiteral() throws LexException {
+    public void testLeadingZero() throws LexException {
```

Comment:
Just one last thing, could you also remove the `throws` declaration, now that there's nothing that should be thrown anymore? 😅

Reply:
For the test to compile, it still needs a `throws` declaration, because we're using checked exceptions in the lexer. The test would still fail if an exception we're to be thrown.